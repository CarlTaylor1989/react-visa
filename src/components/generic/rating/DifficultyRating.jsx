import React from 'react';
import PropTypes from 'prop-types';
import Star from './Star';

import './DifficultyRating.scss';

const DifficultyRating = ({ numberOfStars, currentRating }) => (
  <div className="difficultyRating">
    {[...Array(+numberOfStars).keys()].map(i => (
      <Star
        key={i + 1}
        className="star"
        active={currentRating >= i + 1}
      />
    ))}
  </div>
);

DifficultyRating.defaultProps = {
  numberOfStars: 3
};

DifficultyRating.propTypes = {
  numberOfStars: PropTypes.number,
  currentRating: PropTypes.number.isRequired
};

export default DifficultyRating;
