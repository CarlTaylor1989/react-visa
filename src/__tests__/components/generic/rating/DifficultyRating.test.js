/* global toJson */
import React from 'react';
import DifficultyRating from '../../../../components/generic/rating/DifficultyRating';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    currentRating: 1,
    numberOfStars: 10
  };
});

test('renders the DifficultyRating component', () => {
  const wrapper = shallow(<DifficultyRating {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
