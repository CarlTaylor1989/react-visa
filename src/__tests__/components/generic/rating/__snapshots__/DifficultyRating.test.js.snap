// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DifficultyRating component 1`] = `
<div
  className="difficultyRating"
>
  <StarButton
    active={true}
    className="star"
    key="1"
  />
  <StarButton
    active={false}
    className="star"
    key="2"
  />
  <StarButton
    active={false}
    className="star"
    key="3"
  />
  <StarButton
    active={false}
    className="star"
    key="4"
  />
  <StarButton
    active={false}
    className="star"
    key="5"
  />
  <StarButton
    active={false}
    className="star"
    key="6"
  />
  <StarButton
    active={false}
    className="star"
    key="7"
  />
  <StarButton
    active={false}
    className="star"
    key="8"
  />
  <StarButton
    active={false}
    className="star"
    key="9"
  />
  <StarButton
    active={false}
    className="star"
    key="10"
  />
</div>
`;
